app/views/components/shared/sidebar.rb

```diff
@@ -4,10 +4,22 @@ class Shared::Sidebar < ApplicationComponent
   def view_template
     aside(class: "fixed top-14 z-30 -ml-2 hidden h-[calc(100vh-3.5rem)] w-full shrink-0 md:sticky md:block") do
       div(class: "relative overflow-hidden h-full py-6 pl-8 pr-6 lg:py-8", style: "position:relative;--radix-scroll-area-corner-width:0px;--radix-scroll-area-corner-height:0px") do
-        div(class: "h-full w-full rounded-[inherit]", style: "overflow: hidden scroll;") do
+        div(id: "sidebar-menu", class: "h-full w-full rounded-[inherit]", style: "overflow: hidden scroll;") do
           render Shared::Menu.new
         end
       end
     end
+
+    script do
+      safe <<-JS
+        window.addEventListener("turbo:before-visit", () => {
+          localStorage.setItem("menuScrollPositon", document.getElementById("sidebar-menu").scrollTop);
+        });
+
+        window.addEventListener("turbo:load", () => {
+          document.getElementById("sidebar-menu").scrollTop = localStorage.getItem("menuScrollPositon") || 0;
+        });
+      JS
```

Comment:
I can refactor this code to a stimulus controller. Does it make sense?

Reply:
If you do believe that this code will be around forever, I do think it will be better to refactor it as a stimulus controller. 

If you think this is just a working around until the incompatibility with sticky/overflow/fixed styles could be fixed, I'm fine. If you decide to keep the code, it will be nice to write a comment explaining why it exists.

Reply:
I think it will be around forever, since I could preserve scroll position with turbo but the page content scroll was preserved too, so I'm not sure if turbo is capable to handle well this case.